### PR TITLE
executor: migrate test-infra to testify for tiflash_test.go

### LINKED ISSUE #28621: migrate test-infra to testify for `tiflash_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28621 

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -22,37 +22,30 @@ import (
...
 	"github.com/pingcap/tidb/parser/terror"
 	plannercore "github.com/pingcap/tidb/planner/core"
 	"github.com/pingcap/tidb/session"
 	"github.com/pingcap/tidb/store/mockstore"
 	"github.com/pingcap/tidb/store/mockstore/unistore"
+	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/util/israce"
 	"github.com/pingcap/tidb/util/kvcache"
-	"github.com/pingcap/tidb/util/testkit"
 	"github.com/pingcap/tidb/util/testleak"
```
Comment (tisonkun) [inline executor/tiflash_test.go:42]: Why do we still have this dependency? I think we should finally get rid of it.

File: executor/tiflash_test.go [hunk 2]
```diff
@@ -622,20 +626,22 @@ func (s *tiflashTestSuite) TestDispatchTaskRetry(c *C) {
...
 	tk.MustQuery("select count(*) from t").Check(testkit.Rows("4"))
-	c.Assert(failpoint.Disable("github.com/pingcap/tidb/store/mockstore/unistore/mppDispatchTimeout"), IsNil)
+	require.Nil(t, failpoint.Disable("github.com/pingcap/tidb/store/mockstore/unistore/mppDispatchTimeout"))
 }
 
-func (s *tiflashTestSuite) TestCancelMppTasks(c *C) {
+func TestCancelMppTasks(t *testing.T) {
 	testleak.BeforeTest()
-	defer testleak.AfterTest(c)()
+	defer testleak.AfterTestT(t)()
```
Comment (tisonkun) [inline executor/tiflash_test.go:640]: goroutine leaks checking already covered by `goleak` in `main_test.go`.

File: executor/tiflash_test.go [hunk 3]
```diff
@@ -1109,16 +1135,16 @@ func (s *tiflashTestSuite) TestForbidTiFlashIfExtraPhysTableIDIsNeeded(c *C) {
 		fmt.Fprintf(resBuff, "%s\n", row)
 	}
 	res = resBuff.String()
-	c.Assert(strings.Contains(res, "tiflash"), IsTrue)
-	c.Assert(strings.Contains(res, "tikv"), IsFalse)
+	require.True(t, strings.Contains(res, "tiflash"))
```
Comment (tisonkun) [inline executor/tiflash_test.go:1138]: You may replace with `require.Contains`, ditto others.

File: executor/tiflash_test.go (hunk 2 above)
Comment (feitian124) [reply executor/tiflash_test.go:640]: ok, get it. will fix soon

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution! Merging...

**Pull request merged**